### PR TITLE
Facebook Custom Audiences Destination

### PR DESCRIPTION
**Pull request by bleonard**
Feature Request https://github.com/grouparoo/grouparoo/issues/915

Adds a facebook app and destination for custom audience. 
Lookalike audiences will be in a [future feature request](https://github.com/grouparoo/grouparoo/issues/1107).

### CODE REVIEW

**Comment by bleonard** [comment]
One thing to note is that facebook won't allow you to delete users from an audience if it is fairly small. It's hard to find the right number, but it's 100 or more of actual facebook members.
This makes things somewhat complicated for true syncing.

`Audience Size too low: You cannot remove users from this audience because it will result in a low audience size.` is the error message.

To account for this I've done a few things:
* not let this failure stop syncing to other groups for a user
* Added the option at the destination level to be "Additive" only. That is, don't even try to delete.

**Comment by bleonard** [comment]
Closes T-897

File: plugins/@grouparoo/app-templates/src/destination/shared/batch.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  },
+  config: any
+) {
+  // build up groups situation of group names to addition and removal
+  let oldGroups = exportedProfile.oldGroups || [];
+  let newGroups = exportedProfile.newGroups || [];
+  if (exportedProfile.toDelete) {
+    // if anyone asks, removing from all groups as part of deletion
+    oldGroups = Array.from(new Set(oldGroups.concat(newGroups)));
+    newGroups = [];
```
Comment (evantahler) [inline plugins/@grouparoo/app-templates/src/destination/shared/batch.ts:29]: Think this should be something that core does when `toDelete` is true?

File: plugins/@grouparoo/facebook/src/lib/export/fields.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
+import { GroupSyncModeData } from "@grouparoo/app-templates/dist/destination/group";
```
Comment (evantahler) [inline plugins/@grouparoo/facebook/src/lib/export/fields.ts:1]: This whole file is fascinating.  Can you walk me through it next week?

File: plugins/@grouparoo/facebook/src/lib/export/utils.ts [hunk 3]
```diff
@@ -0,0 +1,332 @@
+/**
```
Comment (evantahler) [inline plugins/@grouparoo/facebook/src/lib/export/utils.ts:1]: What's different here than the facebook lib?  Could we patch it vs re-implemeting?

Comment (bleonard) [reply plugins/@grouparoo/facebook/src/lib/export/utils.ts:1]: for some reason, it was splitting first and last name it to 5 characters. I think there will be a few other changes too.

File: plugins/@grouparoo/app-templates/src/destination/shared/batch.ts (hunk 1 above)
Comment (bleonard) [reply plugins/@grouparoo/app-templates/src/destination/shared/batch.ts:29]: Not sure. For now, I lean on the side of leaving it as it is. knowing which groups they were actually sent (oldGroups) before.

**Pull request merged**